osu.Game.Modes.Osu/UI/OsuPlayfield.cs

```diff
@@ -63,11 +72,69 @@ public override void Add(DrawableHitObject h)
             base.Add(h);
         }
 
+        public override void PostProcess()
+        {
+            AddFollowPoints();
+        }
+
         private void judgement(DrawableHitObject h, JudgementInfo j)
         {
             HitExplosion explosion = new HitExplosion((OsuJudgementInfo)j, (OsuHitObject)h.HitObject);
 
             judgementLayer.Add(explosion);
         }
+
+        public void AddFollowPoints(int startIndex = 0, int endIndex = -1)

```

Comment:
I think this should be abstracted away into a separate class. We want to have better behaviour for other types of follow points than, well, "points".

For example, we want to have proper support for connecting hitobjects with lines instead of points (without the hacks current skins use). These could for instance be skinned with repeating textures or the likes.

I'd suggest making a `FollowConnection` (or similar) class inheriting from `Container` which can then either implement traditional followpoints like you did, or it can just draw things differently depending on implementation.